src/reflect/value.go

```diff
@@ -1035,8 +1035,37 @@ func mulNoOverflow(x, y uintptr) (uintptr, bool) {
 	return m, x == m/y
 }
 
+var zerobuffer [256]byte
+
 func Zero(typ Type) Value {
-	panic("unimplemented: reflect.Zero()")
+	if typ.Size() < unsafe.Sizeof(uintptr(0)) {
+		return Value{
+			typecode: typ.(*rawType),
+			value:    nil,
+			flags:    valueFlagRO,
+		}
+	}
+
+	if typ.Size() < uintptr(len(zerobuffer)) {
+		flags := valueFlagRO
+
+		// slices are bigger than pointer but don't have the indirect flag set
+		if typ.Kind() != Slice {
+			flags |= valueFlagIndirect
+		}
+
+		return Value{
+			typecode: typ.(*rawType),
+			value:    unsafe.Pointer(&zerobuffer[0]),
+			flags:    flags,
+		}
+	}
+
+	return Value{
+		typecode: typ.(*rawType),
+		value:    alloc(typ.Size(), nil),
+		flags:    valueFlagRO | valueFlagIndirect,
```

Comment:
Why is `valueFlagRO` needed? How is it different from not setting either of these flags? (`flags: valueFlag`)? Because that is what would happen when you would use `reflect.ValueOf` on a zero value, like this (see the implementation of `reflect.ValueOf`):

```go
func myZero() reflect.Value {
    var v someType
    return reflect.ValueOf(v)
}
```

I think this would be correct:

```suggestion
		flags:    valueFlagExported,
```

And similarly for the other instances in this function.

Reply:
The idea for `ro` was mostly taken from upstream.  I couldn't off the top of my head figure out a case where the other flags would fail to do what I needed, and it just seemed "safer" to have an explicit flag for "You Can't Touch This."  If you're sure the semantics can be the same using the flags we have, I'll simplify this PR to remove it.